gsms.py

```diff
@@ -205,5 +290,43 @@ def main() -> None:
                                 'custom location.')
 
 
+def get_win_path(folder_id: str) -> str:
+    """
+    Function to resolve Windows UUID folders into their absolute path
+
+    Parameters
+    ----------
+    folder_id : str
+        The folder UUID to convert into the absolute path.
+
+    Returns
+    -------
+    str
+        Resolved Windows path as string.
+
+    Raises
+    ------
+    NotADirectoryError
+        When a UUID can not be resolved to a path as it is not a Windows UUID path this will be raised.
+
+    Examples
+    --------
+    >>> get_win_path("{B4BFCC3A-DB2C-424C-B029-7FE99A87C641}", UserHandle.Common)
+    ...
+    """
+    # Get actual Windows folder id (fid)
+    fid = GUID(uuid.UUID(folder_id))
+    # Prepare pointer to store the path
+    path_pointer = ctypes.c_wchar_p()
+    # Execute function (which stores the path in our pointer) and check return value for success (0 = OK)
+    if _SHGetKnownFolderPath(ctypes.byref(fid), 0, wintypes.HANDLE(0), ctypes.byref(path_pointer)) != 0:
+        raise NotADirectoryError()
```

Comment:
You could leave your message.

`raise NotADirectoryError("The specified UUID could not be resolved to a path")`

Reply:
I put it back but also added the UUID that was passed in